edb/ir/ast.py

```diff
@@ -715,7 +715,7 @@ class MutatingStmt(Stmt):
 
 
 class InsertStmt(MutatingStmt):
-    pass
+    on_conflict: typing.Optional[typing.List[PointerRef]] = None
```

Comment:
SQL `ON CONFLICT` accepts constraint names as conflict targets, let's use a constraint object directly.  I suggest adding a `ConstraintRef` to IR:

```python
class ConstraintRef(ImmutableBase):
    # The id of the constraint
    id: uuid.UUID
    # The module id of the constraint
    module_id: uuid.UUID
```

You can obtain the name of the SQL constraint with `common.get_constraint_backend_name()`

Reply:
That turns out to not get the right constraint name, but I think I figured it out.